app/src/main/java/com/iic/shopingo/ui/HomeActivity.java

```diff
@@ -21,14 +34,61 @@
   @InjectView(R.id.home_action_card_make_request)
   ActionCardView makeRequestCard;
 
+  @InjectView(R.id.home_content_container)
+  View contentContainer;
+
+  @InjectView(R.id.home_spinner_container)
+  View spinnerContainer;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    if (!SharedUserConnector.getInstance().isUserSignedIn()) {
+    init();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    if (CurrentUser.getInstance().state == CurrentUser.State.LOGGED_OUT) {
       navigateToOnboarding();
     } else {
-      // TODO: check user state and go to the correct screen directly.
-      init();
+      showSpinner();
+
+      new GetCurrentState(CurrentUser.getToken()).executeAsync().continueWith(new Continuation<CurrentStateApiResult, Void>() {
+        @Override
+        public Void then(Task<CurrentStateApiResult> task) throws Exception {
+          if (!task.isFaulted() && !task.isCancelled()) {
+            switch (task.getResult().userState) {
+              case LOGGED_OUT: {
+                Log.e("HomeActivity", "Server should not report user as logged out");
+                finish();
+              } break;
+              case IDLE: {
+                // TODO: a user might still have a declined request that he never saw. redirect to the declined request here.
+                showContent();
+              } break;
+              case TRIPPING: {
+                CurrentUser.getInstance().state = CurrentUser.State.TRIPPING;
+                Intent intent = new Intent(HomeActivity.this, ManageTripActivity.class);
+                ArrayList<IncomingRequest> list = new ArrayList<>(task.getResult().activeTripRequests.size());
+                list.addAll(task.getResult().activeTripRequests);
+                intent.putParcelableArrayListExtra(ManageTripActivity.EXTRA_REQUESTS, list);
+                startActivity(intent);
```

Comment:
I may be too optimistic, but I think we'll have time. But this is definitely low priority right now, IMO.